test(prices): rewrite Price List E2E Tests

jniles opened the pull request:
This commit rewrites the price list end to end tests following the page object format.  It also tidies up the HTML a bit throughout the page, including using headercrumbs correctly and bhFilterToggle for toggling
the inline filter on the page.

Closes #3350.

jniles committed:
test(prices): rewrite Price List E2E Tests

This commit rewrites the price list end to end tests following the page
object format.  It also tidies up the HTML a bit throughout the page,
including using headercrumbs correctly and bhFilterToggle for toggling
the inline filter on the page.

Closes #3350.

jeremielodi commented on client/src/modules/prices/modal/create.html:
```diff
@@ -1,39 +1,42 @@
 <div class="modal-header">
-  <h4 ng-if="$ctrl.isCreate" translate>PRICE_LIST.ADD_PRICE_LIST</h4>
-  <h4 ng-if="!$ctrl.isCreate" translate>PRICE_LIST.UPDATE</h4>
+  <ol class="headercrumb">
```
👍 nice

jeremielodi commented on client/src/modules/prices/modal/create.html:
```diff
@@ -1,39 +1,42 @@
     FORM.BUTTONS.CLOSE
   </button>
+
+  <bh-loading-button loading-state="CreateForm.$loading">
```
good use of ```bh-loading-button```

jeremielodi commented on client/src/modules/prices/modal/createItems.html:
```diff
@@ -50,32 +52,37 @@ <h5 class="text-primary">{{ ModalCtrl.pricelist.label }}</h5>
+
+    <div class="row text-right" style="margin-bottom:15px;">
+      <div class="col-md-12">
+        <bh-loading-button loading-state="ModalForm.$loading">
```
👍 This is useful when the internet is slow, ```bh-loading-button ``` prevent to submit twice

jeremielodi commented on client/src/modules/prices/modal/createItems.js:
```diff
@@ -23,7 +23,7 @@ function PriceListItemsModalController(
   vm.cancel = cancel;
   // remove a price list item
-  vm.del = del;
+  vm.remove = remove;
```
This is more readable

jeremielodi commented on client/src/modules/prices/templates/delete_price_item.tmpl.html:
```diff
@@ -1,6 +1,5 @@
-<div class="ui-grid-cell-contents text-center">
-  <a href class="text-danger" ng-click="grid.appScope.del(row.entity.uuid)">
+<div class="ui-grid-cell-contents text-right" data-row="{{row.entity.label}}">
```
good indicators for e2e test

jeremielodi commented on test/integration/weekEndConfig.js:
```diff
@@ -75,7 +75,7 @@ describe('(/payroll/weekend_configuration) The /payroll/weekend_configuration  A
 
   it('DELETE /WEEKEND__CONFIG/:ID will send back a 404 if the WeekEnd Configuration does not exist', () => {
-    return agent.delete('/weekend_config/inknowWeekEnd')
+    return agent.delete('/weekend_config/unknownWeekEnd')
```
👍

jeremielodi reviewed (commented):
Looks good to me

jniles committed:
fix(price list): add missing GROUP BY

This commit fixes a bug where the GROUP BY statement was missing,
leading to price lists without items being filtered out of the registry.
